src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -853,6 +855,51 @@ class WorkspaceService(userInfo: UserInfo, dataSource: DataSource, containerDAO:
       case Some(workflow) => op(workflow)
     }
   }
+
+  private def submitWorkflow(workspaceContext: WorkspaceContext, methodConfig: MethodConfiguration, entity: Entity, wdl: String, authCookie: HttpCookie, txn: RawlsTransaction) : Either[WorkflowFailure, Workflow] = {
+    val inputs = MethodConfigResolver.resolveInputs(workspaceContext, methodConfig, entity, wdl, txn)
+    val successes = inputs collect { case (key, Success(value)) => (key, value) }
+    val failures = inputs collect { case (key, Failure(regret)) => AttributeString(regret.getMessage) }
+    if (failures.isEmpty ) {
+      val execStatus = executionServiceDAO.submitWorkflow(wdl, MethodConfigResolver.propertiesToWdlInputs(successes), authCookie)
+      Right(Workflow(workflowId = execStatus.id, status = WorkflowStatuses.Submitted, statusLastChangedDate = DateTime.now, workflowEntity = AttributeEntityReference(entityName = entity.name, entityType = entity.entityType)))
+    }
+    else {
+      Left(WorkflowFailure(entityName = entity.name, entityType = entity.entityType, errors = failures.toSeq))
+    }
+  }
+
+  private def withSubmissionEntities(submissionRequest: SubmissionRequest, workspaceContext: WorkspaceContext, rootEntityType: String, txn: RawlsTransaction)(op: (Seq[Entity]) => PerRequestMessage): PerRequestMessage = {
+    //If there's an expression, evaluate it to get the list of entities to run this job on.
+    //Otherwise, use the entity given in the submission.
+    if ( submissionRequest.expression.isEmpty ) {
```

Comment:
It would be more consistent with the rest of our code to use ```match``` syntax on Options rather than ```isEmpty``` / ```get```